src/resource/jsonschema/resource.js

```diff
@@ -184,31 +173,21 @@ export default class JSONSchemaResource extends Resource {
         } = options;
         return new JSONSchemaResource({
             id,
-            type: 'object',
             properties: pick(this.properties, properties),
             required: isUndefined(required) ? intersection(this.required, properties) : required,
+            type: 'object',
         });
     }
 
-    /* Add internal definitions.
-     */
-    addDefinitions({ definitions }) {
-        // XXX make definition management nicer
-        this.schema.definitions = {};
-        definitions.forEach(
-            (definition) => {
-                this.schema.definitions[definition.id] = definition;
-            },
-        );
-        return this;
-    }
-
     /* Add another schema to the validator.
      */
-    addSchemaReference({ $ref, schema }) {
+    addReference(schema) {
```

Comment:
This function assumes that references will always to be `#${ID}`, which is false.